test/torch_tests.py

```diff
@@ -0,0 +1,184 @@
+import os
+import sys
+
+import torch
+import torch.distributed as dist
+
+torch.backends.cuda.matmul.allow_tf32 = False
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+from torch.testing._internal.common_distributed import DistTestCases
+from torch.testing._internal.common_utils import (
+    run_tests,
+    sandcastle_skip_if,
+)
+
+# Sets showing that a collective isn't implemented
+DistTestCases.skip_collective["allgather_coalesced"] = {"nccl", "mpi", "ucc"}
+DistTestCases.skip_collective["gather"] = {"nccl", "ucc"}
+DistTestCases.skip_collective["scatter"] = {"nccl", "ucc"}
+DistTestCases.skip_collective["reduce"] = {"ucc"}
+DistTestCases.skip_collective["sendrecv anysource"] = {"nccl", "ucc"}
+DistTestCases.skip_collective["cpu barrier"] = {"nccl", "ucc"}
+
+# Sets showing that something is implemented
+DistTestCases.backend_feature["gpu"] = {"nccl", "gloo", "ucc"}
+DistTestCases.backend_feature["cuda"] = {"nccl", "gloo", "ucc"}
+DistTestCases.backend_feature["ddp"] = {"nccl", "gloo", "ucc"}
+DistTestCases.backend_feature["subgroup"] = {"nccl", "gloo", "ucc"}
```

Comment:
does this mean we run tests on "nccl" and "gloo" backend as well? Wonder why we run those backends on torch-ucc repo.

Reply:
It just means the test would run and not get skipped for those backends. The set doesn't mean that those backends are being used. It probably doesn't matter and the set could be just {"ucc"}, but there is no harm in including them as valid.

Reply:
I will remove mentions to other PGs

Reply:
removed